_posts/2020-03-02-ios-deploys-super-weird-totally-normal.markdown

```diff
@@ -0,0 +1,113 @@
+---
+layout: epic
+title: "iOS Deploys: Super Weird And Totally Normal"
+date: 2020-03-02
+categories: [ios, deployment, continuous deployment, fastlane]
+author: ash
+---
+
+Software deploys! What a concept. You have some code running somewhere, and you need to get it running somewhere
+else. What could possibly go wrong? While web developers have become accustom to some _really slick_ deploy
+processes, iOS developers have to work within some very different constraints.
+
+Today I want to explore the differences between deploying iOS software and front-end/back-end web software. Some of
+these differences are inherent to how the code gets executed, and some of the differences are incidental to choices
+that Apple has made. These are constraints that iOS developers need to work within. As Artsy has adopted React
+Native over the pats four years, we have had more and more of our web engineering colleagues contributing to our
+iOS app. For these web engineers, getting familiar with the iOS deploy constraints is as important as getting to
+know Xcode and CocoaPods.
+
+<!-- more -->
+
+We're going to use a case study to frame today's discussion. Artsy's Mobile Experience team recently got a ticket
+from our Platform team. The nature of the ticket itself doesn't matter, but it involved a change we were making to
+our API. I fixed the bug and submitted a pull request, which quickly got merged. We tested the bug fix in our next
+beta deploy, and everything looked fine. Great so far.
+
+Artsy releases iOS software on a two-week release cadence, and this bug fix was scheduled to be released to users
+the following week. But our back-end team wanted to quickly deploy that API change, which would require the app bug
+fix to get released to users _first_.
+
+So... what to do?
```

Comment:
I like where this question falls - is there a way to make it super super clear exactly what the debate is? E.g. "We had a decision to make: should we deploy off-cadence to get the fix out ASAP, or should we ask the Platform team to be patient and wait until we had a batch of changes to release?"